lib/core/spaces/tuple.ex

```diff
@@ -0,0 +1,16 @@
+defmodule Gyx.Core.Spaces.Tuple do
```

Comment:
what about something like Gyx.Core.Spaces.List or Vector? to imply that any number of items is possible

Reply:
I think these are important decisions. How to name such module, and what data structure use for such implementation. 

The motivation for using `Tuple` as naming is that it follows the convention on  [`Gym`](https://github.com/openai/gym/blob/master/gym/spaces/tuple.py).
In fact, it represents a fixed size, fixed position structure. Any space is always fixed for a particular environment (its action and observation spaces do not change), but they can differ widely among different environments.

The motivation for NOT using an actual tuple as data structure, is that it is not possible to type spec the elements to be a `space :: Gyx.Core.Spaces.Box | Gyx.Core.Spaces.Discrete`
And it might be unnatural to construct a tuple structure from parsing a `Gym` `Tuple` `__repr__` (see [`Gyx.Gym.Utils`](https://raw.githubusercontent.com/doctorcorral/gyx/master/lib/Gym/utils.ex) where this case must be handled).

That's why I vote to name it `Tuple`, having the key `spaces` holding a list.